### PR TITLE
Fix #56 - Folder name issue on windows 

### LINKED ISSUE #56: [BUG] Destination Path issue on windows when subjects name have "/ " in it.
**Context**
* OS: Windows 10
* Install method: scapper exe
* Your location: outside
* Course URL: https://a.impartus.com/ilc/#/course/1424218/706
* Lecture range: 1

**Additional context**
Subject name contains "/" as it is common between two tracks.
"Financial Analytics (S2-20_PDBACZG517 / S2-20_PDFTCZG517)"

**Describe the bug**
Code is deriving the destination folder name from the subject name and since it has forward slash in the name it is breaking.
https://github.com/iamkroot/ilc-scraper/blob/master/ilc_scrape.py#L277

**Screenshots**

**Program output**
<details>
<summary>Click to see output</summary>
<p>

```
Couldn't find C:\Users\<>\Downloads\imp_config.json. Will skip for now.
Traceback (most recent call last):
  File "pathlib.py", line 1273, in mkdir
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Users\\<>\\Documents\\WILP\\lectures\\Financial Analytics (S2-20_PDBACZG517 \\ S2-20_PDFTCZG517) WILP Session'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "ilc_scrape.py", line 347, in <module>
  File "ilc_scrape.py", line 278, in main
  File "pathlib.py", line 1278, in mkdir
  File "pathlib.py", line 1273, in mkdir
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Users\\<>\\Documents\\WILP\\lectures\\Financial Analytics (S2-20_PDBACZG517 \\ S2-20_PDFTCZG517) WILP Session'
```
</p>
</details>

### PR DESCRIPTION
**Pull request by brahma19**
https://github.com/iamkroot/ilc-scraper/issues/56

### CODE REVIEW

File: ilc_scrape.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import requests
 from downloader import DirServer, download_stream
 from utils import print_quit, read_json, sanitize_filepath, sp_args, store_json
+from pathvalidate import sanitize_filename
```
Comment (iamkroot) [inline ilc_scrape.py:14]: The package needs to be added to dependency file using poetry

File: ilc_scrape.py [hunk 2]
```diff
@@ -273,7 +274,7 @@ def main():
         print_quit("No lectures found. Is the url proper?")
 
     total_lecs = len(lectures)
-    subject_name = "{subjectName} {sessionName}".format(**lectures[0])
+    subject_name = sanitize_filename("{subjectName} {sessionName}".format(**lectures[0]))
```
Comment (iamkroot) [inline ilc_scrape.py:277]: The line length should be less than 88 characters. Follow PEP8 convention.
```suggestion
    subject_name = sanitize_filename(
        "{subjectName} {sessionName}".format(**lectures[0])
    )
```

**Review by iamkroot — CHANGES_REQUESTED** [review]
Also see [CONTRIBUTING.md](https://github.com/iamkroot/ilc-scraper/blob/master/CONTRIBUTING.md) for guidelines on how to write proper commit messages.

File: ilc_scrape.py (hunk 1 above)
Comment (brahma19) [reply ilc_scrape.py:14]: Added package to poetry dependency file. 
Bumped the version as well, is that needed ?

File: pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "ilc-scraper"
-version = "0.8.2"
+version = "0.8.3"
```
Comment (iamkroot) [inline pyproject.toml:3]: This should not be a part of this PR. I'll bump the scraper version separately. One PR should only handle one change. Consider what would happen if we had to revert this change - we still wouldn't want to rescind the version change for the scraper.

**Review by iamkroot — CHANGES_REQUESTED** [review]
After making the necessary changes, please squash all your commits into one, and ensure that the commit message follows the `CONTRIBUTING` guidelines.

File: pyproject.toml (hunk 3 above)
Comment (brahma19) [reply pyproject.toml]: Done.

**Review by iamkroot — APPROVED** [review]
LGTM!

**Pull request merged**